rb_ws/src/buggy/scripts/path_planner/path_planner.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python3
+
+import sys
+from threading import Lock
+
+import numpy as np
+import rclpy
+from rclpy.node import Node
+
+from nav_msgs.msg import Odometry
+from std_msgs.msg import Float64
+from buggy.msg import TrajectoryMsg
+
+sys.path.append("/rb_ws/src/buggy/scripts")
+from util.pose import Pose
+from util.trajectory import Trajectory
+
+class PathPlanner(Node):
+    """
+    Class to generate new trajectory splices for SC autonomous system.
+
+    Takes in a default trajectory and an inner curb trajectory.
+
+    """
+
+    # move the curb towards the center of the course by CURB_MARGIN meters
+    # for a margin of safety
+    CURB_MARGIN = 1 #m
+
+    # the offset is calculated as a mirrored sigmoid function of distance
+    OFFSET_SCALE_CROSS_TRACK = 2 #m
+    OFFSET_SCALE_ALONG_TRACK = 0.2
+    ACTIVATE_OTHER_SCALE_ALONG_TRACK = 0.1
+    OFFSET_SHIFT_ALONG_TRACK = 4 #m
+
+    # number of meters ahead of the buggy to generate local trajectory for
+    LOCAL_TRAJ_LEN = 50#m
+
+    # start generating local trajectory this many meters ahead of current position
+    LOOKAHEAD = 2#m
+
+    # number of points to sample along the nominal trajectory
+    RESOLUTION = 150
+
+    # frequency to run in hz
+    FREQUENCY = 10
+
+    def __init__(self) -> None:
+        super().__init__('path_planner')
+        self.get_logger().info('INITIALIZED.')
+
+
+        #Parameters
+        self.declare_parameter("traj_name", "buggycourse_safe.json")
+        traj_name = self.get_parameter("traj_name").value
+        self.nominal_traj = Trajectory(json_filepath="/rb_ws/src/buggy/paths/" + traj_name) #TODO: Fixed filepath, not good
+
+        self.declare_parameter("curb_name", "")
+        curb_name = self.get_parameter("curb_name").value
+        curb_name = None if curb_name == "" else curb_name
+        if curb_name is None:
+            self.left_curb = None
+        else:
+            self.left_curb = Trajectory(json_filepath="/rb_ws/src/buggy/paths/" + curb_name) #TODO: Fixed filepath, not good
+
+        #Publishers
+        self.other_buggy_xtrack_publisher = self.create_publisher(Float64, "self/debug/other_buggy_xtrack", 10)
+        self.traj_publisher = self.create_publisher(TrajectoryMsg, "self/cur_traj", 10)
```

Comment:
do we want to append "/self" to all rostopics pertaining to the buggy's own data? seem unnecessarily verbose to me